span.go

```diff
@@ -105,7 +105,7 @@ func (s *spanImpl) Log(ld opentracing.LogData) {
 	defer s.onLog(ld)
 	s.Lock()
 	defer s.Unlock()
-	if s.trim() {
+	if s.trim() || s.tracer.options.DropAllLogs {
```

Comment:
I wonder why we are holding the lock for this line... did you happen upon the answer? I would hope the options are immutable after tracer construction, and not sure why `s.trim()` would need the lock held either...